src/built-in-apps/tss-bi.js

```diff
@@ -3,21 +3,47 @@ const TssApp = {
   APP_NAME: 'tss-bi',
   useFrost: true,
 
+  onArrive: async function(request) {
+    let {method, data: {params}} = request;
+    switch(method) {
+      case "lock": {
+        let {user, task} = params;
+        let lock = await this.readGlobalMem(`user-lock-${user}`)
+        if (lock) {
+          throw {message: `User [${user}] locked for a moment`}
+        }
+        await this.writeGlobalMem(`user-lock-${user}`, task, 10);
```

Comment:
Is this atomic?
What happens when two nodes call it at the same time?

Reply:
No this part is not atomic.